src/main/java/uk/gov/hmcts/reform/sscs/service/hmc/topic/ProcessHmcMessageService.java

```diff
@@ -50,15 +51,18 @@ public void processEventMessage(HmcMessage hmcMessage)
 
         checkStatuses(caseId, hearingId, hmcMessageStatus, hmcStatus);
 
-        if (stateNotHandled(hmcStatus, hearingResponse)) {
+        final boolean isHearingUpdated = isHearingUpdated(hmcStatus, hearingResponse);
+
+        if (stateNotHandled(isHearingUpdated, hmcStatus, hearingResponse)) {
             log.info("CCD state has not been updated for the Hearing ID {} and Case ID {}",
-                hearingId, caseId);
+                     hearingId, caseId
+            );
             return;
         }
 
         SscsCaseData caseData = ccdCaseService.getCaseDetails(caseId).getData();
 
-        if (isHearingUpdated(hmcStatus, hearingResponse)) {
+        if (isHearingUpdated) {
             hearingUpdateService.updateHearing(hearingResponse, caseData);
```

Comment:
You don't want to call this for a cancelled hearing as it will cause an error; update hearing is only for hearing listed and hearing updated.

Reply:
@TeaWithLucas Do you mean hmcStatus == Cancelled? hmcStatus == Cancelled won't reach this block. All I've done here is a small refactor so isHearingUpdated is called once and the value is stored for use again on this line if it reaches it. This is covered by unit tests.
ListingStatus == CNCL will behave the same as ListingStatus == FIXED as per the jira ticket

Reply:
@marcjudge If ListingStatus == CNCL that means that the hearing is cancelled, if the hearing is cancelled, this method should not be called. This method is only for adding/updating the hearing venue, date etc, when a hearing is cancelled there is no hearing/date, and it will cause an exception. There should be no reason to add ListingStatus == CNCL to this and even if it won't reach this point, it confuses the logic for future development. I hope that explains it.

Reply:
@marcjudge If ListingStatus == CNCL that means that the hearing is cancelled, if the hearing is cancelled, this method should not be called. This method is only for adding/updating the hearing venue, date etc, when a hearing is cancelled there is no hearing/date, and it will cause an exception. There should be no reason to add ListingStatus == CNCL to this and even if it won't reach this point, it confuses the logic for future development. I hope that explains it.

Reply:
@TeaWithLucas That's makes sense to be but it's not inline with the story description - 'We also need to update the ProcessHMCMessageService.java to accept requests from CNCL (as well as FIXED). This should follow the same route. If there are another other places that check for FIXED, these will need to be updated as well.'
Should FIXED and CNCL be treated differently?

Reply:
I've updated this so ListingStatus == CNCL behaves the same as hmcStatus == Cancelled